File: js/설탕 배달(2839).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const N = Number(input[0]);
const dp = Array.from({length: N+1}, () => Infinity);
// 배열 생성 방법 2
// const dp = new Array(N+1).fill(Infinity);

dp[0] = 0;
for (let i=3; i<=N; i++) {
    if (i >= 3 && dp[i-3] !== Infinity) {
        dp[i] = Math.min(dp[i], dp[i-3]+1);
    }
    if (i >= 5 && dp[i-5] !== Infinity) {
        dp[i] = Math.min(dp[i], dp[i-5]+1);
    }
}

console.log(dp[N] === Infinity ? -1 : dp[N]);

File: python/설탕배달(2839).py
# sugar = int(input())

# bag = 0
# while sugar >= 0:
#     if sugar % 5 == 0:
#         bag += (sugar // 5)
#         print(bag)
#         break
#     sugar -= 3
#     bag += 1
# else:
#     print(-1)

sugar = int(input())
dp = [float('inf')] * (sugar + 1)
dp[0] = 0

for i in range(3, sugar+1):
    if i >= 3 and dp[i-3] != float('inf'):
        dp[i] = min(dp[i], dp[i-3]+1)
    if i >= 5 and dp[i-5] != float('inf'):
        dp[i] = min(dp[i], dp[i-5]+1)

print(dp[sugar] if dp[sugar] != float('inf') else -1)
